### PR TITLE
Modify clio to use sbt 1.2, in part by removing some settings.

### PR DESCRIPTION
**Pull request by pshapiro4broad**
Things compile but it's not ready to merge.

This is just an experiment to see how much work would be required to migrate to 1.0.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -4,7 +4,8 @@ import org.broadinstitute.clio.sbt._
 // ## sbt settings
 // For more info, see: https://broadinstitute.atlassian.net/wiki/pages/viewpage.action?pageId=114531509
 
-enablePlugins(GitVersioning)
+lazy val root = (project in file("."))
+  .enablePlugins(GitVersioning)
```
Comment (danxmoran) [inline build.sbt:8]: There's already a root project defined here: https://github.com/broadinstitute/clio/pull/293/files#diff-fdc3abdfd754eeb24090dbd90aeec2ceR51. What's the motivation for scoping the `enablePlugins` on it?

File: project/build.properties [hunk 2]
```diff
@@ -1,3 +1,3 @@
 # Set the sbt version used by the project. scala version set in build.sbt.
 # suppress inspection "UnusedProperty"
-sbt.version=0.13.17
+sbt.version=1.2.1
```
Comment (danxmoran) [inline project/build.properties:3]: I think it's up to 1.2.3 by now

File: project/src/main/scala/org/broadinstitute/clio/integrationtest/ClioIntegrationTestSettings.scala [hunk 3]
```diff
@@ -58,8 +58,9 @@ object ClioIntegrationTestSettings extends BuildInfoKeys {
     Def.task {
       val generatedResourceDir = (resourceManaged in IntegrationTest).value
 
-      val DockerMajorVersionRegex(dockerMajorVersion) =
-        Process(Seq("docker", "--version")).!!.trim
+//      val DockerMajorVersionRegex(dockerMajorVersion) =
+//        Process(Seq("docker", "--version")).!!.trim
```
Comment (danxmoran) [inline project/src/main/scala/org/broadinstitute/clio/integrationtest/ClioIntegrationTestSettings.scala:62]: What's the motivation for this? Did `Process` go away?

File: project/src/main/scala/org/broadinstitute/clio/publish/ArtifactoryFatJarPublishingPlugin.scala [hunk 4]
```diff
@@ -30,9 +30,11 @@ object ArtifactoryFatJarPublishingPlugin extends AutoPlugin {
...
     "[organization]/[module]/[revision]/[type]s/ivy-[revision].xml"
   private val ivyArtifactoryPatterns = Patterns(
-    ivyPatterns = Seq(ivyXmlPattern),
-    artifactPatterns = Seq(ivyArtifactPattern),
-    isMavenCompatible = true
+    ivyPatterns = Vector(ivyXmlPattern),
+    artifactPatterns = Vector(ivyArtifactPattern),
+    isMavenCompatible = true,
+    descriptorOptional = true,
+    skipConsistencyCheck = false
```
Comment (danxmoran) [inline project/src/main/scala/org/broadinstitute/clio/publish/ArtifactoryFatJarPublishingPlugin.scala:37]: Were there equivalent options for `descriptorOptional` and `skipConsistencyCheck` in 0.13 that these are now matching? Or are these brand new?

File: project/src/main/scala/org/broadinstitute/clio/publish/ArtifactoryFatJarPublishingPlugin.scala [hunk 5]
```diff
@@ -49,13 +51,7 @@ object ArtifactoryFatJarPublishingPlugin extends AutoPlugin {
       Seq(
         publishTo := Some(ivyArtifactoryResolver),
         publishMavenStyle := false,
-        isSnapshot := true,
-        moduleSettings := InlineConfigurationWithExcludes(
-          projectID.value,
-          projectInfo.value,
-          Seq.empty,
-          configurations = Seq(Default)
-        )
```
Comment (danxmoran) [inline project/src/main/scala/org/broadinstitute/clio/publish/ArtifactoryFatJarPublishingPlugin.scala:58]: I think we need to keep the `moduleSettings` so gradle can find the jar in the zamboni nightly test.

File: build.sbt (hunk 1 above)
Comment (pshapiro4broad) [reply build.sbt:8]: I forgot but I think `enablePlugins()` no longer worked at the top level

File: project/src/main/scala/org/broadinstitute/clio/publish/ArtifactoryFatJarPublishingPlugin.scala (hunk 5 above)
Comment (pshapiro4broad) [reply project/src/main/scala/org/broadinstitute/clio/publish/ArtifactoryFatJarPublishingPlugin.scala:58]: yeah, I couldn't get it to compile so I removed it for now, but I assumed some stuff will be broken until it's added back in

**Pull request closed without merging**